zapcore/encoder.go

```diff
@@ -168,6 +168,12 @@ func NanosDurationEncoder(d time.Duration, enc PrimitiveArrayEncoder) {
 	enc.AppendInt64(int64(d))
 }
 
+// MillisDurationEncoder serializes a time.Duration to an integer number of
+// milliseconds elapsed.
+func MillisDurationEncoder(d time.Duration, enc PrimitiveArrayEncoder) {
+	enc.AppendInt64(d.Nanoseconds() / 1e6)
```

Comment:
```suggestion
	enc.AppendInt64(d.Milliseconds())
```

Reply:
the d.Milliseconds() is a new API only for go1.13+

Reply:
https://github.com/golang/go/issues/28564

Reply:
Good point, we can migrate to it once we drop support for 1.12. This is good for now, thanks.